apps/dotcom/sync-worker/src/TLUserDurableObject.ts

```diff
@@ -292,7 +303,11 @@ export class TLUserDurableObject extends DurableObject<Environment> {
 					throw new ZMutationError(ZErrorCode.bad_request, `File not found ${nextFileState.fileId}`)
 				}
 				if (file.ownerId === this.userId) return
-				if (file.shared) return
+				if (
+					file.shared &&
+					(nextFileState.userId === this.userId || nextFileState.userId === undefined)
```

Comment:
This also feels like a bug. We should not allow users to change file state for other users.